[Release] 2.0.0 - RC

lgvv opened the pull request:
## 내용
 - 릴리즈 전 불필요한 파일 삭제
 - 릴리즈 전 최종 개발 코드 후보
    - 크리티컬하지 않은 잔여 이슈는 해당 버전 이후에 대응
 - 릴리즈 예정일 5월 첫주


iOS: @x-0o0 @wonniiii 
Android: @mwy3055 @l2hyunwoo 
Server: @zbqmgldjfh

lgvv committed:
[삭제] 릴리즈 전 불필요한 파일 정리

lgvv commented:
> 앱 버전 올려주세요

버전 2.0.0으로 세팅되어 있는데 혹시 어떤 버전일까요??

x-0o0 commented:
> > 앱 버전 올려주세요
> 
> 버전 2.0.0으로 세팅되어 있는데 혹시 어떤 버전일까요??

아 빌드버전이요

x-0o0 commented:
@lgvv 
그리고 다시한번 재차 확인할게요. 4-5번째 확인하는거 같은데
V1 -> v2
로컬데이터 마이그레이션 안해준다는게 솔직히 실무에서 상상도 할 수 없는 거 거든요..??
팀 내 논의가 된게 맞나요? 다같이 논의했는데 결과가 그렇다면 어쩔 수 없는데 전 수차례 말했기 때문에 사용자들 북마크 데이터, 커스텀 알림 동의여부 등 싹다 날라가도 책임 안집니다.

cc. @yeon-kyu @mwy3055 @zbqmgldjfh @wonniiii

x-0o0 commented:
/쿠링 패키지 빌드하고 패키지 테스트

yeon-kyu commented:
> @lgvv 그리고 다시한번 재차 확인할게요. 4-5번째 확인하는거 같은데 V1 -> v2 로컬데이터 마이그레이션 안해준다는게 솔직히 실무에서 상상도 할 수 없는 거 거든요..?? 안하는 것으로 안드가 같이 논의해준 결과 안한다고 하니까 쿠링 외부인인 제가 뭐 어쩔 수 없긴한데 이를 동의한 안드도 솔직히 이해 안되구요, 전 수차례 말했기 때문에 사용자들 북마크 데이터, 추가한 학과 목록 등 싹다 날라가도 책임 안집니다. 배포 후 데이터 마이그레이션 작업하는게 솔직히 소 잃고 외양간 고치기인데 왜 하는지 이해도 안되구요.
> 
> cc. @yeon-kyu @mwy3055 @zbqmgldjfh @ngwoon
> 
> 3기분들은 아직 쿠링 도메인 지식이 없으실테니 2기까지만 태그할게요.

제가 iOS 개발에 대한 정확한 컨텍스트는 없지만 android 의 경우에는 앱을 삭제하고 재설치하거나 앱데이터를 완전히 지우지않는한 계속해서 크래시가 납니다. 아마 iOS도 비슷할것으로 예상하는데 마이그레이션 스킵하려는 이유가 있으실까요? 
사용자에게 매우 안좋은 경험을 줄것으로 예상하는데 스킵하겠다고 한다면 논리적인 설명이 필요할것같습니다

lgvv committed:
[수정] 빌드 버전 규칙에 따라 수정

lgvv commented:
iOS의 경우 관계형DB가 아니고, 비지니스 모델, UI모델, DTO가 전부 하나로 묶여 있습니다. 또한 학과, 일반 대학 부분이 하나로 묶이면서 기존에 학과, 일반에 존재하면 모든 모델을 v2로 들고 들어온 후 `NoticeProvider` 객체로 대응해야 합니다.

서버에서 북마크를 제외하고는 api를 통해 데이터를 내려주는 상황이라 기존 북마크 데이터를 제외하곤 영향 범위도 작으며, 오래된 북마크의 경우에는 중요도도 떨어진다고 판단됩니다.

무엇보다도 더이상 릴리즈 미루기보단 릴리즈 후 효원님이랑 논의가 필요하겠지만, 개인적으로는 일반적으로 널리 사용되는 Realm 도입, 모델 분리 등의 작업을 가져가는게 좋아 보입니다.

데이터 저장 키 값을 변경해서 기존 데이터 유지하고, 충돌 문제 예방하는 방법으로 수정하겠습니다. 

@yeon-kyu @x-0o0

lgvv committed:
[수정] 기존 키값 변경으로 기존 데이터모델과 충돌 가능성 제거

x-0o0 commented:
> iOS의 경우 관계형DB가 아니고, 비지니스 모델, UI모델, DTO가 전부 하나로 묶여 있습니다. 또한 학과, 일반 대학 부분이 하나로 묶이면서 기존에 학과, 일반에 존재하면 모든 모델을 v2로 들고 들어온 후 `NoticeProvider` 객체로 대응해야 합니다.
> 
> 
> 
> 서버에서 북마크를 제외하고는 api를 통해 데이터를 내려주는 상황이라 기존 북마크 데이터를 제외하곤 영향 범위도 작으며, 오래된 북마크의 경우에는 중요도도 떨어진다고 판단됩니다.
> 
> 
> 
> 무엇보다도 더이상 릴리즈 미루기보단 릴리즈 후 효원님이랑 논의가 필요하겠지만, 개인적으로는 일반적으로 널리 사용되는 Realm 도입, 모델 분리 등의 작업을 가져가는게 좋아 보입니다.
> 
> 
> 
> 데이터 저장 키 값을 변경해서 기존 데이터 유지하고, 충돌 문제 예방하는 방법으로 수정하겠습니다. 
> 
> 
> 
> @yeon-kyu @x-0o0 

1. 관계형DB 라는게 왜 언급되는건지 모르겠습니다.
2. 비즈니스 모델이 뭘 의미 하는건지 모르겠습니다. 비즈니스 로직을 말씀하시는건가요.
3. 비즈니스모델, UI모델, DTO가 하나로 묶여있다는게 도대체 무슨 말인지 이해할 수가 없습니다. 코드로 보여주세요.
4. NoticeProvider로 대응하는건 그냥 매핑작업인데 뭐가 문제인가요. 에러를 코드로 보여주세요.
5. 추가한 학과도 서버에서 저장하나요? 북마크도 중요한 데이터인데 너무 가볍게 보는게 아닌가요?
6. 릴리즈를 미루는게 아니라 해야할 작업을 제때 해야하는게 맞는거 아닌가요? 원래 마이그레이션 해도 릴리즈는 미뤄지지 않을 수 있는 충분하고 넉넉한 일정이었습니다

mwy3055 commented:
건우님과 제가 의견을 나눴고, 당시 저의 주장은

1. 복구 불가능한 데이터를 그냥 날려버리면 답이 없다. 나라면 마이그레이션을 할 것.
2. 건우님 설명을 들어보니, 현재 ios 작업 가능한 인원이 건우님밖에 없고, 5월 첫째주로 예상되는 출시 기한까지 마이그레이션 작업을 마치기 어렵다고 이해
3. 그렇다면 ios 팀 의견을 반영하여, 마이그레이션 없이 출시하는 것으로..?

였습니다.

x-0o0 commented:
@lgvv
빌드가 안됩니다. 빌드 되도록 수정 부탁드립니다.

원인 파악에 혹시 문제가 있다면 멘션주세요.

x-0o0 commented on package-kuring/Sources/Caches/Utils/StringSet.swift:
```diff
@@ -6,7 +6,7 @@
 
 class StringSet {
-    static let baseString = "com.kuring.sdk"
+    static let baseString = "com.kuring.sdk.v2"
```
v2 는 왜 붙이셨나요? 이거 붙이면 v1 UserDefaults 참조가 안될텐데 의도가 있으실거 같아 궁금합니다

x-0o0 replied:
컨텍스트 공유 완료 :)

x-0o0 commented:
@lgvv 

> Merge branch 'feature/lgvv/auth' into release/2.0.0

이미 https://github.com/ku-ring/ios-app/pull/204  통해서 develop 에 머지 되었는데 왜 머지하시나요?

에러 원인 파악에 어려움이 있으시다면 알려주세욥~!

lgvv committed:
[추가] 디펜던시 추가

x-0o0 commented:
/쿠링 패키지 빌드

yeon-kyu reviewed (dismissed):
구두로 설명해주신 부분들 이해했습니다. 고생많으셨습니다~!

x-0o0 committed:
[추가] 패키지 타겟 디펜던시 수정

x-0o0 commented:
/쿠링 패키지 빌드

sunshiningsoo committed:
[수정] 앱아이콘 Alternate 리뷰 수정

- 기존 디렉토리에 직접 들고 있는 png 파일 Asset의 AppIcon으로 변환해서 저장해두었습니다.
- plist 파일에서 앱 아이콘 변환하는 것을 pbx/general/ALTERNATE_APPICON_NAMES로 변경합니다.

lgvv commented:
/쿠링 빌드

sunshiningsoo commented:
@x-0o0 
[3e8b8d3](https://github.com/ku-ring/ios-app/pull/205/commits/3e8b8d3935d4f3e1021261e4a233b022aa5c1827) 에서 수정했습니다~! 테플 업로드 한번 부탁드려요!

이전과의 변화로는 아래 두가지입니다
- 디렉토리에서 직접 png 파일 들고 있는 것 -> Asset의 AppIcon으로 변경
- plist에서 Alternate App Icon 관리 -> BuildSetting/Alternate App Icon Sets 에서 관리

lgvv committed:
[수정] 학과 공지 순서 변경

lgvv committed:
[수정] 탭바 이미지 교체 및 본 프로젝트에 광범위하게 적용되는 accentColor를 제거

sunshiningsoo committed:
[수정] 탭바 selection 변경

sunshiningsoo committed:
[수정] tab image single scale로 변경

sunshiningsoo committed:
[수정] 중요 공지 태그 백그라운드 수정

sunshiningsoo committed:
[수정] NoticeRow date 표기법 변경

sunshiningsoo committed:
[수정] NoticeRow 수정

- 자간, 북마크 위치

lgvv committed:
[수정] 앱 아이콘 수정

wonniiii committed:
[수정] BookmarkList 수정

Devider, NoticeRow 타입

lgvv committed:
[수정] 디자이너 아이콘 변경 요청 및 심사 이슈 수정

lgvv commented:
/쿠링 머지

sunshiningsoo reviewed (approved):
머지 후 다시 작업해요!

sunshiningsoo commented:
/쿠링 머지
